flake.nix

```diff
@@ -1,34 +1,54 @@
 {
   inputs.flake-utils.url = github:numtide/flake-utils;
 
-  nixConfig = {
-    # Needed by callCabal2nix
-    allow-import-from-derivation = true;
-  };
-
   outputs = {
     self,
     nixpkgs,
     flake-utils,
   }:
     with flake-utils.lib;
-      eachSystem [system.x86_64-linux] (system: let
+      eachDefaultSystem (system: let
```

Comment:
If I'm not mistaken, the `with flake-utils.lib` was only useful to have both `eachSystem` and `system.x86_64-linux` in the scope. This isn't necessary anymore (and actually things work fine because we shadow the name `system`) so what about going back to the safer:
```nix
flake-utils.lib.eachDefaultSystem (system: ...
```